src/test/java/steps/CreateStep.java

```diff
@@ -0,0 +1,101 @@
+package steps;
+
+import base.BaseUtil;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Then;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import java.text.ParseException;
+
+public class CreateStep extends BaseUtil{
+
+    protected BaseUtil base;
+
+    public CreateStep(BaseUtil base){
+        this.base = base;
+    }
+
+    @And("^I create an all day event$")
+    public void iCreateAnAllDayEventToday() throws InterruptedException {
+        base.wait.until(ExpectedConditions.elementToBeClickable(base.create.nameOfTheEventTextBox));
+        base.create.addNameToEvent(base.prop.getProperty("nameOfTheAllDayEvent")).checkAllDayCheckBox();
+        base.create.addStartDate(base.prop.getProperty("dateOfTheAllDayEvent")).addUntilDate(base.prop.getProperty("dateOfTheAllDayEvent"));
+        base.create.saveEvent();
+    }
+
+    @Then("^All day event should be displayed in the calendar$")
+    public void allDayTodayEventShouldBeDisplayedInTheCalendar() throws Throwable {
+        boolean running = true;
+        while(running) {
+            base.main.clickNextButtonOnWeekView();
```

Comment:
You can use here do {...} while (); I think It is easier to read.

do {
base.main...;
} while (!base.main.startDateIsOnCurrentPage(base.prop.getProperty("dateOfTheAllDayEvent")));